### DISCUSSION TITLE
custom parser

### DISCUSSION THREAD

**Discussion by PhilipBrodovsky** [syntax-tree/unist#94] [Q&A] [answered]
where can I found good example guide or src  (typescript preferred) to create my own **parser (text to AST)**
I try to use unified to compile my own language (similar to HTML)?

thanks for help

**Comment by wooorm** [comment, marked as answer]
There is no guide. See the API docs of unified (https://github.com/unifiedjs/unified#processorparser) and existing code (particularly https://github.com/rehypejs/rehype/tree/main/packages/rehype-parse if you want to parse something close to HTML)!

**Reply by ChristianMurphy** [reply]
Adding on, also see the discussion in https://github.com/orgs/syntax-tree/discussions/36 for more ideas
